test/functional/search_controller_test.rb

```diff
@@ -317,6 +319,18 @@ def no_results
     assert_equal "1",       @response.headers["X-Slimmer-Result-Count"]
   end
 
+  test "should pass the search variant through when one is present" do
+    with_variant SearchMatchLength: "B" do
+      get :index, q: "Variant B"
+    end
```

Comment:
I would assume `with_variant` was only responsible for putting the variant into the request header. Maybe renaming it to something like `with_variant_in_response` would make it clearer that it actually adds an expectation.

Happy either way.